core/src/main/scala/kafka/admin/AdminUtils.scala

```diff
@@ -113,23 +113,31 @@ object AdminUtils extends Logging {
     if (existingPartitionsReplicaList.size == 0)
       throw new AdminOperationException("The topic %s does not exist".format(topic))
 
-    val existingReplicaList = existingPartitionsReplicaList.head._2
+    val existingReplicaListForPartitionZero = existingPartitionsReplicaList.find(p => p._1.partition == 0) match {
+      case None => throw new AdminOperationException("PartitionId should start from 0")
```

Comment:
nit: the error message is a bit misleading here as it seems indicating the adding partitions should start from 0, it should be "the topic does not have partition with id 0".